src/main/scala/io/iohk/ethereum/forkid/ForkIdValidator.scala

```diff
@@ -0,0 +1,129 @@
+package io.iohk.ethereum.forkid
+
+import akka.util.ByteString
+import cats.Monad
+import cats.data.EitherT._
+import cats.implicits._
+import io.iohk.ethereum.utils.BigIntExtensionMethods._
+import io.iohk.ethereum.utils.BlockchainConfig
+import io.iohk.ethereum.utils.ByteUtils._
+import monix.eval.Task
+import org.typelevel.log4cats.Logger
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+import java.util.zip.CRC32
+
+sealed trait ForkIdValidationResult
+case object Connect extends ForkIdValidationResult
+case object ErrRemoteStale extends ForkIdValidationResult
+case object ErrLocalIncompatibleOrStale extends ForkIdValidationResult
+
+object ForkIdValidator {
+
+  implicit val unsafeLogger = Slf4jLogger.getLogger[Task]
+
+  val maxUInt64 = (BigInt(0x7fffffffffffffffL) << 1) + 1 // scalastyle:ignore magic.number
+
+  def validatePeer[F[_]: Monad: Logger](
```

Comment:
Could do with a scaladoc since it's 'external' API, right?

Reply:
Good catch, added

Reply:
Do we really need the monad ?
This is a synchronous function, so there is no need for an async abstraction. I get that logging is a kind of side effect, but from the caller point of view it does not really matter, and it adds some noise because everything needs to be inside a `liftF` now.

Reply:
`Monad` has nothing to do with asynchronicity, it allows chaining calculations,
Logging is a side effect. Are you saying we should ignore that fact?